### ISSUE TITLE
Bug causing "SpecError: Expected externalDocs to be of type ExternalDocumentation with required fields url"

### ISSUE THREAD

**Issue by SassonNir** [Dorthu/openapi3#89]
Hello, after breaking my head for a couple of days trying to figure out why I was getting the aforementioned error saying that the `externalDocs` object could not be parsed.

Finally I found the problem and isolated the case being that the `can_parse` function for the `ExternalDocumentation` object  (which inherits `object_base`'s function) return false even though the object is correct.

Found it to be failing here:
https://github.com/Dorthu/openapi3/blob/7039a15e65477b505467b85bcd09215645f3ec9a/openapi3/object_base.py#L347-L349

This implies that the `url` field is not in the `__slots__` of the object.

Looking at the object (after way too many hours) I noticed that there's a typo:
https://github.com/Dorthu/openapi3/blob/34fb7a062eb5439f2b05b3d2185edb43f398574c/openapi3/general.py#L12-L13

It says `__slos__` instead of `__slots__` and also the `required_fields` needs to be a list.

Thinking that was it, I got another error (again I'm certain that the multiple JSONs that I'm testing on are correct):
```python
SpecError: Expected externalDocs to be one of [<class 'str'>], got <class 'dict'>
```
Which led me to another problem in the `tags` object:
https://github.com/Dorthu/openapi3/blob/34fb7a062eb5439f2b05b3d2185edb43f398574c/openapi3/tag.py#L20
The `externalDocs` object should be either a `dict` or `ExternalDocumentation`.

This also led me to a couple of inconsistencies in several other objects using which have the `externalDocs` field:

https://github.com/Dorthu/openapi3/blob/7039a15e65477b505467b85bcd09215645f3ec9a/openapi3/paths.py#L161
https://github.com/Dorthu/openapi3/blob/7039a15e65477b505467b85bcd09215645f3ec9a/openapi3/openapi.py#L175
https://github.com/Dorthu/openapi3/blob/7039a15e65477b505467b85bcd09215645f3ec9a/openapi3/schemas.py#L95

I'm not sure which one should be (most logical would be `ExternalDocumentation`), I'll be glad to make a pull request fixing it (I've already fixed it on my end and everything works) but I'm not sure which one is correct since leaving everything as is (except the `tag` object since that is almost certainly a bug) doesn't break anything even though I tried breaking it.

Once I get an answer I'll make the pull request.

Thanks.

**Comment by Dorthu** [comment]
Thanks for the detailed bug report.

The errors in the `ExternalDocumentation` class are clearly bugs and should be corrected.  I'm also pretty confident that changing the fields you noted above to be of type `ExternalDocumentation` is correct; looking over the code and related commits, I'm not sure why they aren't already.

If you want to make a PR with the changes you have, I'd appreciate it; otherwise I'll circle back this weekend to make the suggested changes.

**Comment by SassonNir** [comment]
Sure thing, made the pull request #90

**Comment by Dorthu** [comment]
This was shipped in version 1.7.0
